### DISCUSSION TITLE
InvenTree TUI

### DISCUSSION THREAD

**Discussion by j-huff** [inventree/InvenTree#7410] [Show and tell]
Hi all, I've been making a simple terminal user interface (TUI) for InvenTree using the python API and Textual. I'm making this to simplify a few common tasks. So far, the available tabs are "Transfer Items", "Check-In Items", and "Part Search".

It's open source on github: https://github.com/j-huff/inventree-tui

You can also install it with "pip install inventree-tui"

This is the first python package I've published, so forgive me if you try it and it doesn't work. If you do use it and encounter an issue, please post the issue on github!

If you have any suggestions, requests, etc, please let me know either here on preferably as an issue on github. Keep in mind this is intended to be a very simple interface that does not give the user access to most functionality of Inventree.

Planned features:

- Stock taking, adding, counting
- Stock splitting + label printing

If you are interested in contibuting, feel free to fork it the repo and make a pull request. Thanks!

**Comment by matmair** [comment]
Looks interesting. Other toolmakers like @30350n recommend to use `pipx` in their README to ensure that your tool runs in it's own venv - might be interesting to add to yours too.

**Reply by j-huff** [reply]
Thanks, didn't know that was a thing.

**Comment by wolflu05** [comment]
Hey, this looks interesting. Nice that you also added some screenshots to your readme now. That definitely helps to get a first impression for what to expect, before installing and trying it out.
